### PR TITLE
Document all public item in `re_types`

### PR DESCRIPTION
**Pull request by emilk**
### What
This is our public API, and keeping it well-documented seems like a low bar.

In some places it may seem unnecessary (e.g. describing what a `new` function does, or what `Angle::Degrees` means), but enforcing `#[allow(missing_docs)]` is the only way I know to remind us (and other contributors) to always consider documentation.

You can opt-out using `#![allow(missing_docs)]`.

While fixing the docs I found a few weird pieces of code that I also fixed.

### Before
Screenshot 2024-04-29 at 09 04 28
Screenshot 2024-04-29 at 09 04 22

### After
Screenshot 2024-04-29 at 09 03 34
Screenshot 2024-04-29 at 09 03 56

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6146?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6146?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6146)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_types/definitions/rerun/datatypes/angle.fbs [hunk 1]
```diff
@@ -12,9 +12,11 @@ namespace rerun.datatypes;
 union Angle (
   "attr.rust.derive": "Copy, PartialEq"
 ) {
+  /// \rs Angle in radians. One turn is equal to 2π (or τ) radians.
   /// \py 3D rotation angle in radians. Only one of `degrees` or `radians` should be set.
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/datatypes/angle.fbs:16]: the one turn clarifaction is nice, can also go on python.
The fact that python says "3d rotation angle" is also a kinda waiting to be wrong - surely we gonna use the same thing both in 2D!

File: crates/re_types/definitions/rerun/datatypes/transform3d.fbs [hunk 2]
```diff
@@ -12,7 +12,11 @@ namespace rerun.datatypes;
 union Transform3D (
   "attr.rust.derive": "Copy, PartialEq"
 ) {
+  /// Translation plus a 3x3 matrix for scale, rotation, scew, etc.
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/datatypes/transform3d.fbs:15]:
```suggestion
  /// Translation plus a 3x3 matrix for scale, rotation, skew, etc.
```

File: crates/re_types/src/archetypes/view_coordinates_ext.rs [hunk 3]
```diff
@@ -13,65 +14,65 @@ macro_rules! define_coordinates {
...
-    define_coordinates!(RIGHT_HAND_Y_DOWN => (Right, Down, Forward));
-    define_coordinates!(RIGHT_HAND_Z_UP => (Right, Forward, Up));
-    define_coordinates!(RIGHT_HAND_Z_DOWN => (Right, Back, Down));
-    define_coordinates!(LEFT_HAND_X_UP => (Up, Right, Back));
-    define_coordinates!(LEFT_HAND_X_DOWN => (Down, Right, Forward));
-    define_coordinates!(LEFT_HAND_Y_UP => (Right, Up, Forward));
-    define_coordinates!(LEFT_HAND_Y_DOWN => (Right, Down, Back));
-    define_coordinates!(LEFT_HAND_Z_UP => (Right, Back, Up));
-    define_coordinates!(LEFT_HAND_Z_DOWN => (Right, Forward, Down));
+    define_coordinates!("X=Up, Y=Left, Z=Forward", ULF => (Up, Left, Forward));
```
Comment (Wumpf) [inline crates/re_types/src/archetypes/view_coordinates_ext.rs:17]: oof, the macro wasn't able to extract that?

**Review by Wumpf — APPROVED** [review]

File: crates/re_types/src/components/mesh_properties_ext.rs [hunk 4]
```diff
@@ -3,6 +3,7 @@ use crate::datatypes::UVec3D;
 use super::MeshProperties;
 
 impl MeshProperties {
+    /// The triangle indices of the mesh.
```
Comment (jleibs) [inline crates/re_types/src/components/mesh_properties_ext.rs:6]:
```suggestion
    /// Create a new [`MeshProperties`] from an iterable of the triangle indices.
    ///
    /// Each triangle is defined by a UVec3D, where the 3 values are the indices of the 3 vertices
    /// of the triangle.
```

File: crates/re_types/src/tensor_data.rs [hunk 5]
```diff
@@ -320,6 +327,10 @@ pub enum TensorElement {
 }
 
 impl TensorElement {
+    /// Get the value as a 64-bit floating point number.
+    ///
+    /// Note that this may caus rounding for large 64-bit integers,
```
Comment (jleibs) [inline crates/re_types/src/tensor_data.rs:332]:
```suggestion
    /// Note that this may cause rounding for large 64-bit integers,
```

**Review by jleibs — APPROVED** [review]

**Pull request merged**